packages/core/src/utils-hoist/misc.ts

```diff
@@ -22,7 +23,7 @@ interface CryptoGlobal {
  */
 export function uuid4(): string {
   const gbl = GLOBAL_OBJ as typeof GLOBAL_OBJ & CryptoGlobal;
-  const crypto = gbl.crypto || gbl.msCrypto;
+  const crypto = gbl.cryptoTest|| gbl.crypto || gbl.msCrypto;
```

Comment:
This is the only change I want to draw reviewers attention to!

`global.crypto` cannot be modified (likely due to ESM?) so the tests could no longer modify what methods were available. 

Is this a bad idea? Do we need these tests which only cover ancient platforms we don't support?